api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java

```diff
@@ -2541,11 +2543,12 @@ private void enableLabs() {
                 null),
                 null));
 
+        // note that this only only currently accessed via the Lab Workflow "Add Order" button, and the returnUrl and afterAdOrderUrl are both hardcoded below for this
         apps.add(findPatientTemplateApp(Apps.ORDER_LABS,
                 "pih.app.labs.ordering",
                 "icon",
                 Privileges.TASK_ORDER_LABS.privilege(),
-                "/owa/orderentry/index.html?patient={{patientId}}&page=laborders&breadcrumbOverride={{breadcrumbOverride}}&returnUrl=%2F" + WebConstants.CONTEXT_PATH + "%2Fowa%2Flabworkflow%2Findex.html",
+                "/owa/orderentry/index.html?patient={{patientId}}&page=laborders&breadcrumbOverride={{breadcrumbOverride}}&returnUrl=+ " + URLEncoder.encode("/" + WebConstants.CONTEXT_PATH + "/owa/labworkflow/index.html","UTF-8") + "&afterAddOrderUrl=" + URLEncoder.encode("/" + WebConstants.CONTEXT_PATH + "/owa/labworkflow/index.html#/order/{{order}}", "UTF-8"),
```

Comment:
Looks like you've got a stray `+` before the first close quote

Reply:
Thanks, nice catch... I didn't test the returnUrl after refactoring so I missed this.